### PR TITLE
Remove by query with external storage

### PR DESCRIPTION
**Pull request by wmathurin**

### CODE REVIEW

File: libs/SmartStore/SmartStore/Classes/SFSmartStore.m [hunk 1]
```diff
@@ -1975,16 +1975,30 @@ - (void)removeEntriesByQuery:(SFQuerySpec*)querySpec fromSoup:(NSString*)soupNam
...
+        NSString *deleteSql = [NSString stringWithFormat:@"DELETE FROM %@ WHERE %@ in (%@)", soupTableName, ID_COL, limitSql];
+        [self executeUpdateThrows:deleteSql withArgumentsInArray:args withDb:db];
+        // fts
+        if ([self hasFts:soupName withDb:db]) {
+            NSString *deleteFtsSql = [NSString stringWithFormat:@"DELETE FROM %@_fts WHERE %@ in (%@)", soupTableName, ROWID_COL, querySql];
+            [self executeUpdateThrows:deleteFtsSql withDb:db];
+        }
     }
-
-	// TODO Delete External Storage
```
Comment (wmathurin) [inline libs/SmartStore/SmartStore/Classes/SFSmartStore.m]: I didn't see that TODO when reviewing the original PR ;-)

File: libs/SmartStore/SmartStoreTests/SFSmartStoreWithExternalStorageTests.m [hunk 2]
```diff
@@ -287,134 +346,6 @@ - (void)testClearSoupWithExternalStorage {
     }
 }
 
-- (void)testAlterSoupInternalExternalConversion {
```
Comment (wmathurin) [inline libs/SmartStore/SmartStoreTests/SFSmartStoreWithExternalStorageTests.m]: I added tests in SFAlterSoupTests.m for alter soup with storage changes

**Comment by bhariharan** [comment]
👍

**Comment by khawkins** [comment]
👍

**Pull request merged**